test/metabase/native_query_analyzer_test.clj

```diff
@@ -49,7 +49,9 @@
                (q "select \"ID\" from venues"))))
       (testing "you can mix quoted and unquoted names"
         (is (= {:direct #{(mt/id :venues :id) (mt/id :venues :name)} :indirect nil}
-               (q "select v.\"ID\", v.name from venues"))))
+               (q "select v.\"ID\", v.name from venues")))
+        (is (= {:direct #{(mt/id :venues :id) (mt/id :venues :name)} :indirect nil}
+               (q "select v.`ID`, v.name from venues"))))
```

Comment:
Ideally we should have a MySQL/SQLServer-specific driver test to flex the `quotes-preserve-case?` behavior, but that could come in a follow-up PR

Reply:
Good call I should have just included some of those tests in this PR. 

I've got a big 'ol pile of tests coming soon which will cover these. It's a pity I won't be able to use idiomatic SQLServer quotes though 😢